src/libraries/System.Net.Http/src/System/Net/Http/Headers/ProductInfoHeaderValue.cs

```diff
@@ -58,14 +59,15 @@ public override string ToString()
         {
             if (_product == null)
             {
+                Debug.Assert(_comment != null);
```

Comment:
Why is this expected to hold? I presume it's because either _comment or _product must be non-null?

Reply:
yeah, depending on which constructor used one of them will always be set